src/goto-programs/lazy_goto_model.cpp

```diff
@@ -202,13 +202,37 @@ void lazy_goto_modelt::initialize(const cmdlinet &cmdline)
 /// Eagerly loads all functions from the symbol table.
 void lazy_goto_modelt::load_all_functions() const
 {
-  goto_convert(*goto_model, message_handler);
-}
+  std::vector<irep_idt> fn_ids_to_convert;
```

Comment:
Why use a `std::vector` here?

Reply:
I think you might be suggesting that this is declared in the wrong scope, in which case I agree and have fixed it. If you're saying it's the wrong datatype then you'll have to suggest a more suitable alternative because it's not obvious to me.

Reply:
When I wrote that comment I was thinking about the data type (but, yes, the placement was another issue). You are building a queue of unknown size, hence a `std::list` or `std::deque` would seem much more appropriate. What were your reasons for choosing `std::vector` instead?

Reply:
I think `std::vector` is best here. Except for large list items which are expensive to copy, if you don't need `list`'s iterator stability over mutation or `deque`'s efficient push-front operation (which we don't here, since the queue is built once and then drained fully) then `vector` is the highest performance STL sequence container.

Reply:
There are  two reasons I was surprised to see `std::vector` here: 1) If performance were the reason to use a `std::vector`, then I would have expected some use of `reserve` to avoid repeated re-allocation. 2) If performance wasn't the reason, then it must be another unique capability, such as random access, but that isn't used either. Thus a data structure most naturally fitting the demands of a queue of unknown size should be used, IMHO.

Reply:
We could do (1) but we'd need to iterate over the symbol table twice, probably not a saving (or vastly over-reserve, e.g. to symbol_table.size()). Deque and list are both more costly (somewhat and hugely, respectively; http://baptiste-wicht.com/posts/2012/12/cpp-benchmark-vector-list-deque.html). Deque and list offer no advantages for this use case. Therefore we default to `vector`, which is pretty cheap even with re-allocation considering there's only at worst one realloc copy per push-back.

Reply:
Ok, thanks a lot for the pointer!